source/postprocess/tracers.cc

```diff
@@ -93,8 +93,12 @@ namespace aspect
           last_output_time = this->get_time() - output_interval;
         }
 
-      // Advance the particles in the world to the current time
-      world.advance_timestep();
+      // Do not advect the particles in the initial refinement stage
+      const bool in_initial_refinement = (this->get_timestep_number() == 0)
+          && (this->get_pre_refinement_step() < this->get_parameters().initial_adaptive_refinement);
+      if (!in_initial_refinement)
+        // Advance the particles in the world to the current time
+        world.advance_timestep();
```

Comment:
Yes, this looks right -- only do it after the last refinement step. Do I read this right?